tensorboard/components/tf_dashboard_common/data-loader-behavior.ts

```diff
@@ -112,15 +115,14 @@ namespace tf_dashboard_common {
     },
 
     reload() {
-      this._inflightOrLoadedData.clear();
+      this._dataLoadState.clear();
       this._loadData();
     },
 
     reset() {
       // https://github.com/tensorflow/tensorboard/issues/1499
       // Cannot use the observer to observe `loadKey` changes directly.
-      if (this._canceller) this._canceller.cancelAll();
```

Comment:
Hm, if someone calls `reset()` while a datum is still LOADING, can the old request resolve later and trigger `loadDataCallback`, and possibly unwanted `onLoadFinish`?

Maybe we can keep this call in `reset()`, but remove the one in `loadDataImpl()` (as you've done).